### DISCUSSION TITLE
O-RAN NearRT-RIC and xAPP Tutorial

### DISCUSSION THREAD

**Discussion by doogie05** [srsran/srsRAN_Project#817] [General Help]
I was able to follow instructions of O-RAN NearRT-RIC and xAPP tutorial and complete all steps successfully.
I added some debug code in lib/phy, and rebuilt the source with cmake command as specified in the tutorial.

The tutorial uses 'docker compose up 5gc' to bring up Open5GS Core.
When I do this, I don't see debug code I added in lib/phy.

How can I bring up Open5GS from the source I built in /build folder?
Please comment.

Thanks.
James

**Comment by pgawlowicz** [comment]
Hi,
Did you modify open5gs or srsgnb code?

**Reply by doogie05** [reply]
no, only lib/phy/lower/modulation files only.
Thanks

**Reply by pgawlowicz** [reply]
`docker compose up 5gc` starts Open5gc core network in a docker container, I do not understand what it has to do with the changes you made in lib/phy?

**Reply by doogie05** [reply]
I believe there is a "disconnect".
When docker compose up 5gc starts Open5gc core network, does it start the container from the locally built images or pull from the pre-built images (presumably by srsRAN) from docker repo?
Looking at Dockerfile and docker-compose.yml, I thought it was pulling the pre-built images rather than using images I built locally.  Please comment.  Thanks.

**Reply by pgawlowicz** [reply]
If you execute `docker compose up gnb` it is going to build the binary using local version of the code. 
Could you try with `docker compose up --build gnb` to force rebuilding?

**Reply by pgawlowicz** [reply]
@doogie05 any update on this issue?
